kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -93,6 +94,16 @@ public TransactionBuilder withTableProperties(Engine engine, Map<String, String>
     return this;
   }
 
+  /**
```

Comment:
I know we already discussed this but I actually think if it's for a testing API, a direct method on `TransactionImpl` makes more sense, since we basically want to "hack in" and set the `domainMetadatas` field to something specific. Where as for future features, we will be setting `domainMetadatas` inline in the code execution within `TransactionImpl`.

But no need to block on this, we can change it in the future when implementing row tracking.